main.ts

```diff
@@ -1,339 +1,377 @@
-            win.setVisibleOnAllWorkspaces(false);
-        }
-    }
-});
+  }
+})
 
-hook.register(ipcMain, event => send(event), () => {
+hook.register(
+  ipcMain,
+  (event) => send(event),
+  () => {
     dialog.showErrorBox(
-        'Iohook is required to run PoE Overlay',
-        'Iohook could not be loaded. Please make sure you have vc_redist installed and try again.');
-    app.quit();
-});
+      'Iohook is required to run PoE Overlay',
+      'Iohook could not be loaded. Please make sure you have vc_redist installed and try again.'
+    )
+    app.quit()
+  }
+)
 
 /* general */
 
-ipcMain.on('app-version', event => {
-    const version = app.getVersion();
-    event.returnValue = version;
-});
+ipcMain.on('app-version', (event) => {
+  event.returnValue = app.getVersion()
+})
 
 /* changelog */
 
-function showChangelog() {
-    const changelog = new BrowserWindow({
-        modal: true,
-        parent: win,
-    });
-    changelog.removeMenu();
-    changelog.loadURL('https://github.com/PoE-Overlay-Community/PoE-Overlay-Community-Fork/blob/master/CHANGELOG.md#Changelog');
+function showChangelog(): void {
+  const changelog = new BrowserWindow({
+    modal: true,
+    parent: win,
+  })
+  changelog.removeMenu()
+  changelog.loadURL(
+    'https://github.com/PoE-Overlay-Community/PoE-Overlay-Community-Fork/blob/master/CHANGELOG.md#Changelog'
+  )
 }
 
 /* main window */
 function createWindow(): BrowserWindow {
-    const { bounds } = getDisplay();
-
-    // Create the browser window.
-    win = new BrowserWindow({
-        width: bounds.width,
-        height: bounds.height,
-        x: bounds.x,
-        y: bounds.y,
-        transparent: true,
-        frame: false,
-        resizable: false,
-        movable: false,
-        webPreferences: {
-            nodeIntegration: true,
-            allowRunningInsecureContent: serve,
-            webSecurity: false
-        },
-        focusable: false,
-        skipTaskbar: true,
-        show: false
-    });
-    win.removeMenu();
-    win.setIgnoreMouseEvents(true);
-
-    win.setAlwaysOnTop(true, 'pop-up-menu', 1);
-    win.setVisibleOnAllWorkspaces(true);
-
-    win.once('show', () => {
-        if (state.isVersionUpdated(app.getVersion())) {
-            showChangelog();
-        }
-    })
+  const { bounds } = getDisplay()
+
+  // Create the browser window.
+  win = new BrowserWindow({
+    width: bounds.width,
+    height: bounds.height,
+    x: bounds.x,
+    y: bounds.y,
+    transparent: true,
+    frame: false,
+    resizable: false,
+    movable: false,
+    webPreferences: {
+      nodeIntegration: true,
+      allowRunningInsecureContent: serve,
+      webSecurity: false,
+    },
+    focusable: false,
+    skipTaskbar: true,
+    show: false,
+  })
+  win.removeMenu()
+  win.setIgnoreMouseEvents(true)
+
+  if (process.platform !== 'linux') {
+    win.setAlwaysOnTop(true, 'pop-up-menu', 1)
+  } else {
+    win.setAlwaysOnTop(true)
+  }
```

Comment:
I see this code repeats from line 135 above. Should this be pulled out into a helper function that's called from each place?

Reply:
I don't think so

The first initial call to this is on app init and the other specific one on game update which sets different parameters